sql/modules/Reconciliation.sql

```diff
@@ -30,11 +30,17 @@ BEGIN;
 CREATE OR REPLACE FUNCTION reconciliation__submit_set(
         in_report_id int, in_line_ids int[]) RETURNS bool AS
 $$
+DECLARE
+        result NUMERIC;
 BEGIN
-        UPDATE cr_report set submitted = true where id = in_report_id;
-        PERFORM reconciliation__save_set(in_report_id, in_line_ids);
-
-        RETURN FOUND;
+        result = reconciliation__check_balanced(in_report_id);
+        IF result = 0 THEN
```

Comment:
Why are you checking balancedness *before* adding the new line ids to the report?

Reply:
The entries are already added and a submit is asked for an unbalanced report.